src/main/java/org/mskcc/cbio/oncokb/service/SlackService.java

```diff
@@ -576,43 +578,34 @@ private ButtonElement buildApproveButton(UserDTO user) {
                 trialGroup.add(buildEmailOption(DropdownEmailOption.GIVE_TRIAL_ACCESS, user));
             }
             // Add option - Convert to regular
-            if (trialAccountActivated) {
+            if (user.isActivated() && trialAccountActivated) {
                 trialGroup.add(buildConvertToRegularAccountOption(user));
             }
             if (!trialGroup.isEmpty()) {
                 optionGroups.add(OptionGroupObject.builder().label(PlainTextObject.builder().text("Trial").build()).options(trialGroup).build());
             }
-
-            // Add option group - License
-            List<OptionObject> licenseGroup = new ArrayList<>();
-            for (DropdownEmailOption licenseOption: Arrays.stream(DropdownEmailOption.values()).filter(mo -> mo.getCategory() == EmailCategory.LICENSE).collect(Collectors.toList())) {
-                licenseGroup.add(buildEmailOption(licenseOption, user));
-            }
-            optionGroups.add(OptionGroupObject.builder().label(PlainTextObject.builder().text("License").build()).options(licenseGroup).build());
         }
 
+        // Add other option groups
         if (!user.isActivated()) {
-            // Add option group - Clarify
-            List<OptionObject> clarifyGroup = new ArrayList<>();
-            for (DropdownEmailOption clarifyOption : Arrays.stream(DropdownEmailOption.values()).filter(mo -> mo.getCategory() == EmailCategory.CLARIFY).collect(Collectors.toList())) {
-                clarifyGroup.add(buildEmailOption(clarifyOption, user));
-            }
-            optionGroups.add(OptionGroupObject.builder().label(PlainTextObject.builder().text("Clarify").build()).options(clarifyGroup).build());
-
-            // Add option group - Deny
-            List<OptionObject> denyGroup = new ArrayList<>();
-            for (DropdownEmailOption denyOption : Arrays.stream(DropdownEmailOption.values()).filter(mo -> mo.getCategory() == EmailCategory.DENY).collect(Collectors.toList())) {
-                denyGroup.add(buildEmailOption(denyOption, user));
+            for (EmailCategory emailCategory : Arrays.stream(EmailCategory.values()).filter(ec -> !ec.equals(EmailCategory.TRIAL)).collect(Collectors.toList())) {
+                List<OptionObject> optionGroup = new ArrayList<>();
+                for (DropdownEmailOption emailOption : Arrays.stream(DropdownEmailOption.values()).filter(eo -> eo.getCategory().equals(emailCategory)).collect(Collectors.toList())) {
+                    if (emailOption.getSpecificLicenses().isEmpty() || emailOption.getSpecificLicenses().contains(user.getLicenseType()))
+                        optionGroup.add(buildEmailOption(emailOption, user));
+                }
+                if (!optionGroup.isEmpty()) {
+                    optionGroups.add(OptionGroupObject.builder().label(PlainTextObject.builder().text(emailCategory.getLabel()).build()).options(optionGroup).build());
```

Comment:
by giving EmailCategory a label property we can generalize option group creation since each dropdown option has an EmailCategory. trial has its own because there is different logic with conversion to regular and giving trial access